### ISSUE TITLE
cut / copy / paste     special keyboards layout, shortcut

### ISSUE THREAD

**Issue by Enhakiel** [Microsoft/vscode#37549]
Hello,

I've got a "special" keyboard, bépo layout, my ctrl+c and ctrl+v are quite unusable with one hand on the keyboard and the other one on mouse, I've got shortcuts (built in the keyboard to emulate cut/copy/paste) , I assume dvorak may be the same, my keyboard's shortcuts work right in the editor but I cannot use them in the vs code explorer to copy/paste files…

Regards.

**Comment by alexandrudima** [comment]
@Enhakiel You haven't mentioned what OS you're using. Have you seen https://github.com/Microsoft/vscode/wiki/Keybinding-Issues

**Comment by Enhakiel** [comment]
Now I've read it but it doesn't help.
I'm using both Linux and Windows.

![bepo](https://user-images.githubusercontent.com/6618150/32742424-b31cb1d2-c8a9-11e7-8876-0b8a37c75d99.jpg)

This may help you to understand…  the red ones are the ones which work in the editor but not in the explorer, the blue ones work all the time but are really not convenient…  when I said "the vs code explorer" I meant the "app/folder structure" ctrl+shift+e

**Comment by krux02** [comment]
can you change the tite again and make it not bépo layout specific. My keyboard layout (de neo) has a special undo key, with the exact same problem. I can't bind it to anything in vscode.

**Comment by nicolas-goudry** [comment]
To anyone using bépo layout on Linux, the fix is to add this to your parameters:

`"keyboard.dispatch": "keyCode"`

Restart VSCode and it will work like a charm!

**Comment by Enhakiel** [comment]
doesn't work for me :s

**Comment by nicolas-goudry** [comment]
@Enhakiel did you try the fix I gave? If yes, did you restart VSCode?

I’m using VSCode 1.24.0 on Ubuntu 18.04.

**Comment by Enhakiel** [comment]
@nicolas-goudry yes for both, I'm using latest versions of VSCode standard and insider. I'm on Debian 9.4
Have you set TypeMatrix layout with dpkg-reconfigure keyboard-configuration ? I've tried different compose key locations but without success…

**Comment by graph** [comment]
I use dvorak and I also have the issue where the shortcuts don't work properly. I will try the keycode trick and see if it fixes my problem next time I'm on linux/ubuntu.

**Comment by sboulkour** [comment]
Heya ! I too am using a secondary layout and facing this issue.
It works fine when typing code, but **while I hold the ctrl key, layout is switched to my primary layout**. And when I release the ctrl key its back to my secondary layout.

My setup is similar to @Enhakiel : TypeMatrix keyboard, using azerty + bépo layouts in Ubuntu (18.04)

**Comment by Spirarel** [comment]
> To anyone using bépo layout on Linux, the fix is to add this to your parameters:
> 
> `"keyboard.dispatch": "keyCode"`
> 
> Restart VSCode and it will work like a charm!

Worked for me!

I'm on,
MacOS 12.1.1
VSC      1.35.1

**Comment by TeXitoi** [comment]
The copy, cut, paste on the typematrix send [IBM CUA shortcuts](https://en.wikipedia.org/wiki/IBM_Common_User_Access) i.e.:
 - The Cut is Shift+Del;
 - Copy is Ctrl+Ins;
 - Paste is Shift+Ins;

So these common shortcut must not be implemented in the vscode explorer.

**Comment by bgrieder** [comment]
VsCode 1.59 - Ubuntu 21.04 - TypeMatrix BÉPO keyboard 
The fix `"keyboard.dispatch": "keycode"` has stopped working.
My copy and paste keys are now unusable (cut key still work though)
Any change in keyboard keycode handling in 1.59 ?
